src/core/threadcontext.h

```diff
@@ -208,6 +208,9 @@ struct MVMThreadContext {
     /* Last payload made available in a payload-goto exception handler. */
     MVMObject *last_payload;
 
+    /* Handler the currently active CONTROL exception is unwinding to. */
+    MVMActiveHandler *active_control_handler;
```

Comment:
MVMActiveHandler isn't a perfect fit. We don't need MVMActiveHandler->next and we save the payload in MVMActiveHandler->ex_obj, which is a poor name. Should we just create an MVMActiveControlHandler solely for this?

Reply:
Probably better to have a different structure. Also, one needs to GC mark the object.